service_storagefacility/app/services/StorageNodeService.scala

```diff
@@ -663,19 +672,39 @@ class StorageNodeService @Inject() (
     mid: MuseumId,
     oids: Seq[ObjectId]
   ): Future[MusitResult[Seq[ObjectsLocation]]] = {
-    localObjectDao.currentLocations(oids).flatMap { objNodeMap =>
-      val nodeIds = objNodeMap.values.flatten.toSeq.distinct
-      unitDao.getNodesByIds(mid, nodeIds).flatMap { nodes =>
-        Future.sequence {
-          nodes.map { node =>
-            unitDao.namesForPath(node.path).map { namedPath =>
-              val objects = objNodeMap.filter(_._2 == node.id).keys.toSeq
-              // Copy node and set path to it
-              ObjectsLocation(node.copy(pathNames = Option(namedPath)), objects)
-            }
-          }
-        }.map(MusitSuccess.apply)
-      }
+    localObjectDao.currentLocations(oids).flatMap {
+      case MusitSuccess(objNodeMap) =>
+        val nodeIds = objNodeMap.values.flatten.toSeq.distinct
+
+        def findObjectLocations(
```

Comment:
This isn't a good place to define a function, since it's basically defined inside the flatMap of a future. It's better to place it as the first thing in the surrounding method. And add another argument for `objNodeMap`. See the changes I pushed.